File: app/src/main/java/mega/privacy/android/app/interfaces/UploadBottomSheetDialogActionListener.java
package mega.privacy.android.app.interfaces;

/**
 * This interface is to define what methods the activity
 * should implement when having UploadBottomSheetDialog
 */
public interface UploadBottomSheetDialogActionListener {
    void uploadFiles();

    void uploadFolder();

    void takePictureAndUpload();

    void scanDocument();

    void showNewFolderDialog();

    void showNewTextFileDialog(String typedName);
}

File: app/src/main/java/mega/privacy/android/app/interfaces/UploadBottomSheetDialogActionListener.kt
package mega.privacy.android.app.interfaces

/**
 * This interface is to define what methods the activity
 * should implement when having UploadBottomSheetDialog
 */
interface UploadBottomSheetDialogActionListener {
    fun uploadFiles()
    fun uploadFolder()
    fun takePictureAndUpload()
    fun scanDocument()
    fun showNewFolderDialog(typedText: String? = null)
    fun showNewTextFileDialog(typedName: String? = null)
}